lua/orgmode/org/text_objects.lua

```diff
@@ -140,4 +140,132 @@ function TextObjects.around_subtree_from_root()
   current_subtree_from_root(false)
 end
 
+---@class PairedTokenLocations
+---@field content string The content that was detected for the token
+---@field sindex integer The 0-indexed, starting position in the string the first token was found at
+---@field eindex integer The 0-indexed ending position in the string the second token was found at
+
+---Gets all positions of paried tokens
+---@param str string The string to search
+---@param token string The pair of strings to search for
+---@param strict? boolean Should not allow reuse of end token from previous pair for new pair, default: true
+---@param init? integer Starting position to search from in the string, default: 1
+---@return PairedTokenLocations locations The paired token locations
+function TextObjects.get_paired_token_locations(str, token, strict, init)
```

Comment:
Sure, I separated it because I felt it was a bit easier to understand and thought it might be applicable elsewhere.

I can inline it when I get back.